src/oop/special_class_method/medium/q13/MainTest.java

```diff
@@ -0,0 +1,161 @@
+            _assertEquals(value2, testClass.getFieldValue(testInstance, attributeName2),
+                    String.format(modifiedClassMessage, attributeName2));
+        }
+    }
+
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(432.3, "myCompany"),
+                Arguments.of(54.3, "Best Det"),
+                Arguments.of(123.3, "clearer"),
+                Arguments.of(65.4, "Get Clean"),
+                Arguments.of(234.4, "Good Detergent")
+        );
+    }
+
+    private static Stream<Arguments> updateStringInputProvider() {
+        return Stream.of(
+                Arguments.of(34.3, "myCompany", "Best Det"),
+                Arguments.of(432.2, "Best Det", "Get Clean"),
+                Arguments.of(5.5, "clearer", "Good Detergent"),
+                Arguments.of(23.3, "Get Clean", "myCompany"),
+                Arguments.of(78.7, "Good Detergent", "clearer")
+
+        );
+    }
+
+    @Test
+    public void getManufacturerIsDefinedCorrectly() {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(getAttributeMethodName2, null), String.format(incorrectMethodDefinition, getAttributeMethodName2, className));
+        assertTrue(testClass.hasModifier(getAttributeMethodName2, null, "public"), String.format(incorrectModifierMessage, getAttributeMethodName2, className));
+        assertTrue(testClass.hasReturnType(getAttributeMethodName2, null, String.class), String.format(incorrectReturnType, getAttributeMethodName2, className));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void correctGetManufacturerMethod(double value1, String value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, double.class},
+                {value2, String.class}
+        };
+        Object testInstance = testClass.createInstance(arguments);
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(getAttributeMethodName2, null), String.format(incorrectMethodDefinition, getAttributeMethodName2, className));
+        assertTrue(testClass.hasModifier(getAttributeMethodName2, null, "public"), String.format(incorrectModifierMessage, getAttributeMethodName2, className));
```

Comment:
Please place the checks that the method is defined correctly before the call to the constructor in `createInstance()`